### PR TITLE
Bert finetuning

### LINKED ISSUE #21: Add an example script for doing GLUE evaluation of a trained BERT model
Essentially, we need to fill in the section of the README here:
https://github.com/keras-team/keras-nlp/tree/master/examples/bert#evaluating-bert-with-glue

This will explicitly not be adding anything to the library proper, but will help us better evaluate what modular components we can abstract out for running fine-tuning and inference with GLUE.

### PR DESCRIPTION
**Pull request by mattdangerw**
Adds a script for finetuning BERT.

Fixes #21

### CODE REVIEW

File: examples/bert/run_pretraining.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def get_config(self):
         return config
 
 
-class MaskedLM(keras.layers.Layer):
+class MaskedLMHead(keras.layers.Layer):
```
Comment (chenmoneygithub) [inline examples/bert/run_pretraining.py:145]: Curious - why are we renaming it to MaskedLMHead?

**Review by chenmoneygithub — COMMENTED** [review]
Thanks for the PR! Generally looks good, left one comment on renaming.

File: examples/bert/run_pretraining.py (hunk 1 above)
Comment (mattdangerw) [reply examples/bert/run_pretraining.py:145]: I just wanted to get in the habit of naming all these layer that attach to transformers ouputs for loss function "heads." So you have MaskedLMHead and ClassificationHead in this file. I think the uniformity in naming will help make the example clearer.

Not that this is head as in a "screwdriver head," a different piece you attach to a standard tool. I think it's common usage when discussing these models.

File: examples/bert/run_glue_finetuning.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+
+def pack_inputs(
+    inputs,
+    seq_length,
+    start_of_sequence_id,
+    end_of_segment_id,
+    padding_id,
+):
+    # In case inputs weren"t truncated (as they should have been),
```
Comment (fchollet) [inline examples/bert/run_glue_finetuning.py:83]: weren't

File: examples/bert/run_glue_finetuning.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    )
+    # Pad to dense Tensors.
+    input_word_ids, _ = tftext.pad_model_inputs(
+        segments_combined, seq_length, pad_value=padding_id
+    )
+    input_type_ids, input_mask = tftext.pad_model_inputs(
+        segment_ids, seq_length, pad_value=0
+    )
+    # Assemble nest of input tensors as expected by BERT model.
+    return dict(
```
Comment (fchollet) [inline examples/bert/run_glue_finetuning.py:102]: Use {}

File: examples/bert/run_glue_finetuning.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    validation_ds = to_tf_dataset(data["validation" + test_suffix])
+    return train_ds, test_ds, validation_ds
+
+
+class BertClassificationFinetuner(keras.Model):
+    """Adds a classification head to a pre-trained BERT model for finetuning"""
+
+    def __init__(self, bert_model, hidden_size, num_classes, **kwargs):
+        super().__init__(**kwargs)
+        self._bert_model = bert_model
```
Comment (fchollet) [inline examples/bert/run_glue_finetuning.py:151]: Nit: no underscore should be preferred (attributes that the match names of the keyword arguments)

**Review by fchollet — APPROVED** [review]

File: examples/bert/run_glue_finetuning.py (hunk 2 above)
Comment (mattdangerw) [reply examples/bert/run_glue_finetuning.py]: done!

File: examples/bert/run_glue_finetuning.py (hunk 3 above)
Comment (mattdangerw) [reply examples/bert/run_glue_finetuning.py]: done

File: examples/bert/run_glue_finetuning.py (hunk 4 above)
Comment (mattdangerw) [reply examples/bert/run_glue_finetuning.py]: done
